src/main/cpp/controllers/TurretController.cpp

```diff
@@ -113,8 +113,10 @@ void TurretController::Update(units::second_t dt, units::second_t elapsedTime) {
                          ToVector2d(m_turretNextPoseInGlobal.Translation()));
     units::radian_t drivetrainNextThetaInGlobal =
         m_drivetrainNextPoseInGlobal.Rotation().Radians();
-    units::radian_t turretDesiredHeadingInDrivetrain =
+    units::radian_t turretDesiredHeadingInGlobal =
         turretThetaToTargetInGlobal - drivetrainNextThetaInGlobal;
+    units::radian_t turretDesiredHeadingInDrivetrain =
+        turretDesiredHeadingInGlobal - kR;
 
     SetGoal(turretDesiredHeadingInDrivetrain, 0_rad_per_s);
```

Comment:
```suggestion
    units::radian_t turretDesiredHeadingInDrivetrain =
        turretThetaToTargetInGlobal - drivetrainNextThetaInGlobal;
    units::radian_t turretDesiredHeadingInTurret =
        turretDesiredHeadingInDrivetrain - kR;

    SetGoal(turretDesiredHeadingInTurret, 0_rad_per_s);
```
To explain this suggested change, I'd rather keep `turretDesiredHeadingInGlobal` on line 116 as `turretDesiredHeadingInDrivetrain`. `InGlobal` isn't strictly correct since the angle is no longer measured from the edge of the field when `drivetrainNextThetaInGlobal` is subtracted. It's really measured from what the drivetrain considers the front, and that makes it the drivetrain frame. You're correct that we need the angle in the turret frame instead of the drivetrain frame though for the turret controller.

The transformation at lines 118-119 converts `turretDesiredHeadingInDrivetrain` from the drivetrain frame to the turret frame, so I suggest naming the variable the result is assigned to `turretDesiredHeadingInTurret`.